mava/components/training/model_updating.py

```diff
@@ -295,24 +294,23 @@ def model_update_epoch(
 
             base_key, shuffle_key = jax.random.split(key)
 
-            # TODO (dries): This assert is ugly. Is there a better way to do this check?
-            # Maybe using a tree map of some sort?
-            # shapes = jax.tree_util.tree_map(
-            #         lambda x: x.shape[0]==trainer.store.full_batch_size, batch
-            #     )
-            # assert ...
-            assert (
-                list(batch.observations.values())[0].observation.shape[0]
-                == trainer.store.full_batch_size
-            )
+            # Note (dries): This computation is only performed once at compilation tome
+            # and not at every epoch step.
+            batch_shape = jax.tree_util.tree_leaves(
+                list(batch.observations.values())[0].observation
+            )[0].shape[0]
```

Comment:
Can leave it here, but might be cleaner to store the batch size in the store as we used to do with full_batch_size

Reply:
Agreed. I removed it as it is the same size as sample_batch_size :+1: